flycheck.el

```diff
@@ -7647,9 +7647,36 @@ Return the absolute path to the directory"
   "Return directory where rebar.config is located."
   (locate-rebar3-project-root buffer-file-name))
 
+(flycheck-def-option-var flycheck-erlang-rebar3-profile nil erlang-rebar3
+  "The rebar3 profile to use.
+
+The profile used when compiling, if VALUE is nil `test' will be used
+when the file is located in test directory, otherwise `default' will be
+used as profile."
+  :type '(choice
+          (const :tag "Guess rebar3 profile" nil)
+          (const :tag "Use default rebar3 profile" default)
+          (const :tag "Use test rebar3 profile" test))
+  :safe (lambda (value) (or (not value) (memq value '(default test))))
+  :package-version '(flycheck . "32"))
+
+(defun flycheck-erlang-rebar3-get-profile ()
+  "Return rebar3 profile.
+
+Use flycheck-erlang-rebar3-profile if set, otherwise use test profile if
+dirname is test or else default."
+  (cond
+   (flycheck-erlang-rebar3-profile
+    (symbol-name flycheck-erlang-rebar3-profile))
+   ((and buffer-file-name
+         (string= "test" (file-name-base (directory-file-name
+                                          (file-name-directory buffer-file-name)))))
```

Comment:
You want to line feed after `"test"` and reindent in order to pass the `make check` test.